### PR TITLE
Graphslam devel 2merge

### PR DESCRIPTION
**Pull request by bergercookie**
**Changed apps/libraries:**
mrpt-graphslam

Pull-Request contains the new graphslam-engine library and corresponding application files.
In addition PR contains typo fixes as well as explanatory comments in various parts of the MRPT libraries/applications.
PR also adds a new MRPT sample _dirs_files_manipulation_ which is focused on manipulating files and directories using the corresponding MRPT system:: methods (creaateDirectory, renameFile etc.)

PR addresses issue #2 of MRPT/GSoC2016-discussions.

I acknowledge to have: 
* Read the [`CONTRIBUTING`](https://github.com/MRPT/mrpt/blob/master/.github/CONTRIBUTING.md) page
* Updated [`docs/doxygen-pages/changelog.h`](https://github.com/MRPT/mrpt/blob/master/doc/doxygen-pages/changeLog_doc.h) to describe these changes (if applicable)
* Updated [`AUTHORS`](https://github.com/MRPT/mrpt/blob/master/AUTHORS) (if applicable)

(Notify: @MRPT/owners, @jlblancoc, @EduFdez   )

### CODE REVIEW

File: apps/graphslam-engine/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+# Dependencies on MRPT libraries:
+#  Just mention the top-level dependency, the rest will be detected automatically,
+#  and all the needed #include<> dirs added (see the script DeclareAppDependencies.cmake for further details)
+DeclareAppDependencies(${PROJECT_NAME}
+	mrpt-graphslam
+	mrpt-graphs
+	mrpt-gui
+	mrpt-obs
+	mrpt-maps
+	mrpt-hwdrivers
```
Comment (jlblancoc) [inline apps/graphslam-engine/CMakeLists.txt]: Why does this app depend on hwdrivers?? Was it a typo?

File: doc/man-pages/CMakeLists.txt [hunk 2]
```diff
@@ -26,7 +26,7 @@ IF (PROG_GZIP AND PROG_POD2MAN)
 		add_dependencies(man_pages_all man_page_${appname})
 		if (UNIX)
 			INSTALL(FILES "${MRPT_BINARY_DIR}/man-pages/${appname}.1.gz" DESTINATION ${mrpt_apps_INSTALL_PREFIX}share/man/man1/ )
-		endif()
+		endif(UNIX)
```
Comment (jlblancoc) [inline doc/man-pages/CMakeLists.txt]: No need to change this, but just FYI: the verbose CMake notation of `if(XXX) ... endif(XXX)` is sort of a relic from very old CMake versions, so new CMake code shouldn't have it. In fact, for MRPT 2.0.0 one of the pending tasks will be make a clean up of those redundant expressions ;-)

File: doc/man-pages/CMakeLists.txt [hunk 3]
```diff
@@ -61,6 +61,7 @@ IF (PROG_GZIP AND PROG_POD2MAN)
 	CREATE_MANPAGE_PROJECT(hmt-slam-gui)
 	CREATE_MANPAGE_PROJECT(track-video-features)
 	CREATE_MANPAGE_PROJECT(graph-slam)
+	CREATE_MANPAGE_PROJECT(graphslam-engine)
```
Comment (jlblancoc) [inline doc/man-pages/CMakeLists.txt]: I love that you even created the manpage... great!!! :+1:

File: libs/graphslam/CMakeLists.txt [hunk 4]
```diff
@@ -1,12 +1,19 @@
...
-define_mrpt_lib_header_only(
+define_mrpt_lib(
 	# Lib name
 	graphslam
 	# Dependencies
-	mrpt-graphs
 	mrpt-base
+	mrpt-graphs
+	mrpt-gui
+	mrpt-hwdrivers
```
Comment (jlblancoc) [inline libs/graphslam/CMakeLists.txt]: Again, why the hwdrivers here??

**Comment by jlblancoc** [comment]
Amazing, and *huge* GSoC 2016 contribution... thanks!! :+1: 
I left a few comments for you within the code, mostly minor stuff. Will merge the PR anyway in the meanwhile, feel free of continuing doing more commits + PRs as needed.

File: apps/graphslam-engine/CMakeLists.txt (hunk 1 above)
Comment (bergercookie) [inline apps/graphslam-engine/CMakeLists.txt]: hmm, at some point I think that I was using one of the methods/classes in that library...
I will look it up and delete them if they are not necessary.

**Comment by bergercookie** [comment]
Wow, thanks @jlblancoc.
Feels good to be an MRPT developer.. :)

**Pull request merged**